Conch/lib/Conch/Controller/DatacenterRack.pm

```diff
@@ -0,0 +1,146 @@
+=pod
+
+=head1 NAME
+
+Conch::Controller::DatacenterRack
+
+=head1 METHODS
+
+=cut
+
+package Conch::Controller::DatacenterRack;
+
+use Mojo::Base 'Mojolicious::Controller', -signatures;
+use Conch::UUID 'is_uuid';
+
+use Conch::Models;
+
+
+=head2 under
+
+Supports rack lookups by uuid and name
+
+=cut
+
+sub under ($c) {
+	my $r;
+
+	if($c->param('id') =~ /^(.+?)\=(.+)$/) {
+		if($1 eq 'name') {
+			$r = Conch::Model::DatacenterRack->from_name($2);
+		} else {
+			$c->status(404 => { error => "Not found" });
+			return undef;
+		}
+	} else {
+		$r = Conch::Model::DatacenterRack->from_id($c->param('id'));
+	}
+
+	if ($r) {
+		$c->stash('rack' => $r);
+		return 1;
+	} else {
+		$c->status(404 => { error => "Not found" });
+		return undef;
+	}
+}
+
+=head2 create
+
+=cut
+
+sub create ($c) {
+	return $c->status(403) unless $c->is_global_admin;
+	my $d = $c->validate_input('RackCreate') or return;
+
+	unless(Conch::Model::DatacenterRoom->from_id($d->{datacenter_room_id})) {
+		return $c->status(400 => { "error" => "Room does not exist" });
+	}
+
+	unless(Conch::Model::DatacenterRackRole->from_id($d->{role})) {
+		return $c->status(400 => { "error" => "Rack role does not exist" });
+	}
+		
+	my $r = Conch::Model::DatacenterRack->new($d->%*)->save();
+
+	$c->status(303 => "/rack/".$r->id);
+
+}
+
+=head2 get
+
+Get a single rack
+
+=cut
+
+sub get ($c) {
```

Comment:
For both get and get_all, should restrict to global admin.

You can also pull this logic into `under`, which allows you to de-dup the check for update and delete

Reply:
Fixed.